.contrib/Harvesters/WoWTools Harvester/Harvester.py

```diff
@@ -22,193 +24,170 @@
+            build_list.write(build + "\n")
+
+
+def get_thing_ids(thing: Things, build: str) -> list[str]:
+    """Get the IDs of a thing from a build."""
+    thing2table = {
+        Things.Achievements: "achievement",
+        Things.Factions: "faction",
+        Things.FlightPaths: "taxinodes",
+        Things.Followers: "garrfollower",
+        Things.Illusions: "transmogillusion",
+        Things.Mounts: "mount",
+        Things.Pets: "battlepetspecies",
+        Things.Quests: "questv2",
+        Things.Recipes: "skilllineability",
+        Things.Titles: "chartitles",
+        Things.Toys: "toy",
+        Things.Transmog: "itemmodifiedappearance",
+    }
+    url = f"https://wow.tools/dbc/api/export/?name={thing2table[thing]}&build={build}"
     headers = {
         "User-Agent": "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.76 Safari/537.36"
     }
-    csv_name = thing + ".csv"
-    tools_list = []
+    thing_list = list[str]()
     request = requests.get(url, headers=headers)
-    with open(csv_name,"wb") as csv_file:
-        csv_file.write(request.content)
-    with open(csv_name, "r") as csv_file:
-        reader = csv.DictReader(csv_file)
-        for row in reader:
-            if thing == "Illusions":
-                tools_list.append(row["SpellItemEnchantmentID"]+"\n")
-            elif thing == "Mounts":
-                tools_list.append(row["SourceSpellID"]+"\n")
-            elif thing == "Titles":
-                tools_list.append(row["Mask_ID"]+"\n")
-            elif thing == "Toys":
-                tools_list.append(row["ItemID"]+"\n")
-            #elif thing == "Recipes:"
-            else: #"Achievements", "Factions", "Flight Paths", "Followers", "Pets", "Quests", "Transmog":
-                tools_list.append(row["ID"]+"\n")
-    return tools_list
-
-# This function takes the input(thing from thing_list) as searches in Categories.lua for every ID responding to that thing and adds them in a file.
-def get_categories_IDs(thing):
-    path = "C:\\Program Files (x86)\\World of Warcraft\\_retail_\\Interface\\AddOns\\AllTheThings\\db\\Categories.lua"
-    categories_list = []
-    with open(path, "r") as categories_file:
-        for line in categories_file.readlines():
-            line = line.split(",")
-            for ID in line:
+    reader = csv.DictReader(request.content.decode("utf-8").splitlines())
+    for row in reader:
+        match thing:
+            case Things.Illusions:
+                thing_list.append(row["SpellItemEnchantmentID"] + "\n")
+            case Things.Mounts:
+                thing_list.append(row["SourceSpellID"] + "\n")
+            case Things.Titles:
+                thing_list.append(row["Mask_ID"] + "\n")
+            case Things.Toys:
+                thing_list.append(row["ItemID"] + "\n")
+            case _:
+                thing_list.append(row["ID"] + "\n")
```

Comment:
I made it like this and not another dict `thing2column` only because @gbrage said Recipes will need 2 columns.

Reply:
Yes it will need row["SkillLine"] and row["Spell"] So we know where to sort it. 
The Profession Dict is the corresponding SkillLineID. I will add some more common ones